src/lib/controllers/history-controller.js

```diff
@@ -35,15 +39,16 @@ export default class HistoryController {
    *   if there is no active alignment group but there exists saved alignment groups, then we would activate previous group
    */
   undo () {
-    if (this.alignment.hasActiveAlignment && this.alignment.activeAlignmentGroup.groupLen > 1) {
-      return this.alignment.undoInActiveGroup()
-    }
-    if (this.alignment.hasActiveAlignment && this.alignment.activeAlignmentGroup.groupLen === 1) {
-      return this.alignment.undoActiveGroup()
-    }
-    if (!this.alignment.hasActiveAlignment && this.alignment.alignmentGroups.length > 0) {
-      return this.alignment.activateGroupByGroupIndex(this.alignment.alignmentGroups.length - 1)
+    let result
+    if (this.alignment.hasActiveAlignmentGroup && this.alignment.activeAlignmentGroup.groupLen > 1) {
+      result = this.alignment.undoInActiveGroup()
+    } else if (this.alignment.hasActiveAlignmentGroup && this.alignment.activeAlignmentGroup.groupLen === 1) {
+      result = this.alignment.undoActiveGroup()
+    } else if (!this.alignment.hasActiveAlignmentGroup && this.alignment.alignmentGroups.length > 0) {
+      result = this.alignment.activateGroupByGroupIndex(this.alignment.alignmentGroups.length - 1)
     }
+    this.store.commit('incrementAlignmentUpdated')

```

Comment:
I am not sure that there could be such cases, because if we could do nothing this action won't be executed.
But I think that it would be good to add a condition to limit store.commit execution.
Thank you for the note.